### ISSUE TITLE
Is it possible to use this library to stream a large audio/video file, without downloading the whole file?

### ISSUE THREAD

**Issue by mitchdowney** [RonRadtke/react-native-blob-util#147]
I'm trying to add caching for mp3/mp4s before playing them in our media player, and I got react-native-blob-util setup, but it appears to wait until the whole file is downloaded before resolving. Is there an existing way to make RNBU work for this purpose?

**Comment by RonRadtke** [comment]
Currently no since a promise / callback can only resolve once and this is done after completing the download.
For your usecase the promise would have to resolve directly after creating the file cache so you can read the chunks manually out of the file.

**Comment by mitchdowney** [comment]
Got it. Thanks. I'm using the react-native-track-player library, and I learned it already has caching built in for Android, but not iOS. I'm considering working on adding caching for iOS in that library, however if react-native-blob-util could handle the use case I imagine a lot of people would want to use that feature.

https://github.com/doublesymmetry/react-native-track-player/issues/883

I'm not sure if it's feasibly possible in react-native-blob-util though, as it might need implementation details specific to the media player we're using. Feel free to close this issue if it's out of scope for react-native-blob-util.
